src/olympia/blocklist/utils.py

```diff
@@ -97,26 +96,25 @@ def block_activity_log_delete(obj, deleted, *, submission_obj=None, delete_user=
         else amo.LOG.BLOCKLIST_BLOCK_DELETED
     )
 
-    if submission_obj:
-        details['signoff_state'] = submission_obj.SIGNOFF_STATES.for_value(
-            submission_obj.signoff_state
-        ).display
-        if submission_obj.signoff_by:
-            details['signoff_by'] = submission_obj.signoff_by.id
+    details['signoff_state'] = submission_obj.SIGNOFF_STATES.for_value(
+        submission_obj.signoff_state
+    ).display
+    if submission_obj.signoff_by:
+        details['signoff_by'] = submission_obj.signoff_by.id
 
     log_create(
         *[action, *([obj.addon] if obj.addon else []), obj.guid, obj],
         details=details,
-        user=submission_obj.updated_by if submission_obj else delete_user,
+        user=submission_obj.updated_by,
     )
     log_create(
         amo.LOG.BLOCKLIST_VERSION_UNBLOCKED,
         *((Version, version_id) for version_id in changed_version_ids),
         obj,
-        user=obj.updated_by,
+        user=submission_obj.updated_by,
```

Comment:
This is the fix: `obj` here is the `Block` instance we're deleting - we want the `BlocklistSubmission` that holds the information about what/who/why we're deleting instead, and that's `submission_obj`. The rest of the code was already doing that correctly.